src/Handler/Workspace/Create/CreateWorkspaceHandler.php

```diff
@@ -127,9 +125,49 @@ public function __invoke(
         $publicPart = json_encode($keyData);
         assert($publicPart !== false);
 
+        $this->waitUntilBindingsPropagate($cloudResourceManager, $projectName, $wsServiceAcc->getEmail());
         return (new CreateWorkspaceResponse())
             ->setWorkspaceUserName($publicPart)
             ->setWorkspacePassword($privateKey)
             ->setWorkspaceObjectName($dataset->id());
     }
+
+    private function waitUntilBindingsPropagate(
+        Google_Service_CloudResourceManager $cloudResourceManager,
+        string $projectName,
+        string $wsServiceAccEmail
+    ): void {
+        $retryPolicy = new SimpleRetryPolicy(5);
+        $backOffPolicy = new ExponentialBackOffPolicy();
+
+        $proxy = new RetryProxy($retryPolicy, $backOffPolicy);
+        $proxy->call(function () use ($cloudResourceManager, $projectName, $wsServiceAccEmail): void {
+            $actualPolicy = $cloudResourceManager->projects->getIamPolicy($projectName, (new GetIamPolicyRequest()));
+            $actualPolicy = $actualPolicy->getBindings();
+
+            $serviceAccRoles = [];
+            foreach ($actualPolicy as $policy) {
+                if (in_array('serviceAccount:' . $wsServiceAccEmail, $policy->getMembers())) {
+                    $serviceAccRoles[] = $policy->getRole();
+                }
+            }
+
+            $expected = [
+                IAmPermissions::ROLES_BIGQUERY_DATA_VIEWER, // readOnly access
+                IAmPermissions::ROLES_BIGQUERY_JOB_USER,
+            ];
+
+            sort($expected);
+            sort($serviceAccRoles);
+
+            // ws service acc must have a job user role to be able to run queries
+            assert(
+                $expected === $serviceAccRoles,
+                sprintf(
+                    'SA has incorrect roles assigned: %s',
+                    implode(',', $serviceAccRoles)
+                )
+            );
```

Comment:
Ta hláška je na prd docela, představsi, že to na tebe vyskočí v ops-erros.

```suggestion
            if ($expected !== $serviceAccRoles) {
                throw new RuntimeException(sprintf(
                    'Workspace service account has incorrect roles. Expected roles: [%s], actual roles: [%s]',
                    implode(',', $expected),
                    implode(',', $serviceAccRoles)
                ));
            }
```